targetSdk 35: sign up & login dark mode support fixes

mzorz opened the pull request:
⚠️ Works in conjuction with https://github.com/Simperium/simperium-android/pull/251
Pending:
- [x] update the commit SHA for simperium (done in https://github.com/Automattic/simplenote-android/pull/1743/commits/84953ab7b80c2d550901f0776c16e88e30d1c94e)
- [x] update the commit to a tagged version once approved

### Fix
- addressing this issue https://github.com/Automattic/simplenote-android/issues/1741 for which we needed to make updates to the underlying auth activities.
- making the "System default" the default theme preference


Login:
| Login - Before                                                                                                                     | Login - After                                                                                                                      | Login - After - button enabled                                                                                                     |
|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="591" height="1280" alt="image" src="https://github.com/user-attachments/assets/d8b80472-4069-48a8-a9f5-85b0ea12c3bf" /> | <img width="367" height="710" alt="image" src="https://github.com/user-attachments/assets/ad0990da-767a-4b60-8eb5-04c462fe82d8" /> | <img width="347" height="719" alt="image" src="https://github.com/user-attachments/assets/3df3f779-2022-4d26-a42c-93ee3db66e7c" /> |


Sign up:

| Signup - Before                                                                                                                    | Signup - After                                                                                                                     | Signup - After - button enabled                                                                                                    |
|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="332" height="709" alt="image" src="https://github.com/user-attachments/assets/faf6416b-8ef5-4632-9df9-42593347f068" /> | <img width="365" height="722" alt="image" src="https://github.com/user-attachments/assets/c103a3e8-9aa8-4f2d-83eb-f73f831a6b96" /> | <img width="348" height="722" alt="image" src="https://github.com/user-attachments/assets/643a166b-71af-43cc-bd25-35f236ba7739" /> |


### Test
1. set up dark mode on your phone
2. install the app
3. launch it
4. see the login screen looks fine
5. see the sign up screen looks fine

**Caveat**: I see the primary button has a blue background with black text when enabled, I think it should be white text, but for now I'm respecting the defined styles. We can change that if we feel it's better
<img width="289" height="53" alt="image" src="https://github.com/user-attachments/assets/3110d0a1-cf65-4eec-ab9b-a92779230c3a" />

Looking forward to feedback from review.

### Review
<!--
***(Required)*** Add instructions for reviewers.  For example:
Only one developer and one designer are required to review these changes, but anyone can perform the review.
-->

### Release
<!--
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
`RELEASE-NOTES.txt` was updated in d3adb3ef with:
> Added markdown support
-->
<!--
If the changes should not be included in release notes, add a statement to this section. For example:
These changes do not require release notes.
-->

mzorz committed:
fixed sign up and login dark / light temming  made compatible with Android 15 and prior

mzorz committed:
Merge branch 'mzorz/update-targetsdk-35' into mzorz/update-targetsdk-35-sign-up-login-dark-mode

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mzorz committed:
updated simperium commit hash

mzorz committed:
changing default theme to follow system settings

mzorz committed:
apply edge-to-edge and adjust margins on enter password screen

mzorz committed:
clearing prefs for theme upon logout

mzorz committed:
removing clearing the theme prefs on logout, better to stick to defaults

mzorz committed:
adjusted background and style for toolbar on enter password screen

mzorz committed:
removed AUTO theme since MODE_NIGHT_AUTO_TIME is deprecated - still keeping the switch value if it was kept in preferences so it doesnt break for old users

mzorz commented on Simplenote/src/main/java/com/automattic/simplenote/utils/ThemeUtils.java:
```diff
@@ -66,16 +66,14 @@ public static void setTheme(Activity activity) {
                 AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                 break;
+            case THEME_AUTO:
             case THEME_SYSTEM:
```
keeping THEME_AUTO since it can be saved to prefs if you're coming from a previous version of the app, although we removed support for [MODE_NIGHT_AUTO_TIME](https://developer.android.com/reference/androidx/appcompat/app/AppCompatDelegate#MODE_NIGHT_AUTO_TIME()) since it's deprecated.

theck13 replied:
Good idea for backwards compatibility.  It's sad that setting is deprecated.  Pixels have the [Dark Theme scheduling feature](https://support.google.com/pixelphone/answer/7169926), which will work the same way if the ***System default*** theme is selected in ***Settings*** of Simplenote.

Twinsen81 commented:
Tested again, the fix works, thank you! But I've spotted two more issues with the dark mode:
1. On the start screen select `LOG IN`.
2. Enter an email that is not registered on Simplenote yet.
3. The code entering screen's `ENTER PASSWORD` button has no background (or rather has the wrong color, as in the light mode it's just a clickable black text, not a button with a background):

<img width="391" height="1280" alt="image" src="https://github.com/user-attachments/assets/8540aa2e-f337-40ee-a847-54dd66f519ce" />

1. On the start screen select `SIGN UP`.
2. Enter an email that is not registered on Simplenote yet.
3. The `email sent` text doesn't have enough contrast:

<img width="391" height="1280" alt="image" src="https://github.com/user-attachments/assets/013bd622-bea9-4dfb-a69c-fd71e3b80523" />

theck13 commented:
Good catch, @Twinsen81!  There are more than a few things to update like [those you mentioned above](https://github.com/Automattic/simplenote-android/pull/1743#issuecomment-3163218906).  I'm waiting for https://github.com/Simperium/simperium-android/pull/251 to merge before making a final review pass and submitting my suggestions.

mzorz committed:
changed colors for text visibility in email sent screens

mzorz committed:
updated commit hash

mzorz commented:
Thanks for the feedback folks! 

Addressed and updated the simperium branch, and also fixed the findings by @Twinsen81 in 5124e4f643118ec753c0b99114040fe38d3f4597

Dark mode:

| Before                                                                                                                  | After                                                                                                                 |
|-------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="591" height="1280" alt="image" src="https://github.com/user-attachments/assets/0188e179-ff9d-4289-a6cc-a952ebc0d6a6" /> | <img width="356" height="743" alt="image" src="https://github.com/user-attachments/assets/08946677-2694-492f-a744-ce8f9918e1f5" /> |


Light mode:
| Before                                                                                                                             | After                                                                                                                              |
|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="370" height="748" alt="image" src="https://github.com/user-attachments/assets/66f658b7-408b-4336-af0d-b32a8bdfc0f0" /> | <img width="370" height="748" alt="image" src="https://github.com/user-attachments/assets/66f658b7-408b-4336-af0d-b32a8bdfc0f0" /> |



**### second issue**


**Light mode:**
| Before                                                                                                                             | After                                                                                                                              |
|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="368" height="764" alt="image" src="https://github.com/user-attachments/assets/37de526c-d5ac-4dcf-a92b-ad8c56c25ad3" /> | <img width="351" height="753" alt="image" src="https://github.com/user-attachments/assets/30a817f6-0306-4a4e-8a5b-9dd9a120d780" /> |


**Dark mode:**

| Before                                                                                                                             | After                                                                                                                              |
|------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| <img width="360" height="737" alt="image" src="https://github.com/user-attachments/assets/e3201ddd-94b5-4bfd-b4a1-f946b63f50f5" /> | <img width="359" height="745" alt="image" src="https://github.com/user-attachments/assets/cc1db87b-e175-40fb-898d-32839e4823b8" /> |

Twinsen81 reviewed (approved):
LGTM! The dark mode issues has been fixed. Thank you, @mzorz!

mzorz committed:
updated simperium library version

theck13 commented on Simplenote/src/main/res/xml/preferences.xml:
```diff
@@ -43,7 +43,7 @@
         <ListPreference
             android:key="pref_key_theme"
-            android:defaultValue="0"
+            android:defaultValue="2"
```
Smart thinking making the ***System default*** theme the default theme.  That way the logged out and in screens have the same theme on first install.

theck13 reviewed (commented):
There are still some things we can update to optimize the authentication screens (i.e. login and signup).  I'll submit the proposed changes in a subsequent pull request.

Edit:  The subsequent pull request is https://github.com/Automattic/simplenote-android/pull/1747.
